tests/test.py

```diff
@@ -30,6 +30,31 @@ def test_multi_line_condition():
     assert not _results(s)
 
 
+# GOOD (use parentheses for line continuation)
```

Comment:
I've compiled a hand-full of examples for you

```python
# GOOD (have all brackets on the same line)
a = [1, 2, 3]

# GOOD
# (when the opening bracket is the last thing on a line, the matching closing
# bracket is the first thing on a new line that matches the indentation of the
# line with the opening bracket)
a = [
    1, 2, 3
]

# BAD (opening bracket is last, but closing is not on new line)
a = [
    1, 2, 3]

# BAD
# (opening bracket is last, closing is on new line but indentation mismatch)
a = [
    1, 2, 3
    ]

# GOOD (opening bracket is not last, so we don't care about the closing one)
a = [1, 2, 3
     ]

# GOOD (opening bracket is not last, so we don't care about the closing one)
a = [1, 2, 3
]


# (pretty much the same rules apply for multiple brackets)

# GOOD (have all brackets on the same line)
a = [[1, 2, 3], [4, 5, 6]]

# GOOD
a = [
    [1, 2, 3], [4, 5, 6]
]

# GOOD
a = [
    [1, 2, 3],
    [4, 5, 6]
]

# GOOD
a = [[
    1, 2, 3
]]

# GOOD
a = [
    [
        1, 2, 3
    ],
    [
        4, 5, 6
    ]
]

# BAD
# (opening bracket is last, closing is on new line but indentation mismatch)
a = [
    [
        1, 2, 3
    ],
    [
        4, 5, 6
    ]
  ]

# BAD
# (opening bracket is last, closing is on new line but indentation mismatch)
a = [
    [
        1, 2, 3
     ],
    [
        4, 5, 6
    ]
]

# BAD
# (opening bracket is last, closing is on new line but indentation mismatch)
a = [
    [
        1, 2, 3
    ],
    [
        4, 5, 6
     ]
]

# BAD (if you have a closing bracket on a new line, don't open a new bracket)
a = [[
    1, 2, 3
], [
    4, 5, 6
]]


# The same rules should also apply for 
# curly brackets (`{`, `}`) and parenthesis (`(`, `)`)
```

Reply:
Hint: don't only think of lists, dict, sets, and tuples, but expressions (uniary ops, bin ops, etc.), too.